drivers: modem: sim7080: Fix two issues detected at build time in CI

aescolar opened the pull request:
drivers: modem: sim7080: Fix invalid call to strtok_r
    
    The second parameter to strtok_r cannot be null.
    It is the separator strtok is meant to search for, and it is
    not kept anywhere in the saved state (the saved state
    is just a pointer to the next piece of the original string).
    

    drivers: modem: sim7080: Fix missing prototype of strtok_r
    
    strtok_r is a POSIX extension of string.h. To ensure this C library
    header provides the prototype we need to define _POSIX_C_SOURCE.
    So let's do so.
    
    drivers: modem: sim7080: Remove orphan file
    
    drivers/modem/simcom-sim7080.c is not built.
    And the author indicates it was added to the tree in error.
    Let's remove it.

Related #94929

aescolar committed:
drivers: modem: sim7080: Fix missing prototype of strtok_r

strtok_r is a POSIX extension of string.h. To ensure this C library
header provides the prototype we need to define _POSIX_C_SOURCE.
So let's do so.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
drivers: modem: sim7080: Fix invalid call to strtok_r

The second parameter to strtok_r cannot be null.
It is the separator strtok is meant to search for, and it is
not kept anywhere in the saved state (the saved state
is just a pointer to the next piece of the original string).

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

lgehreke reviewed (changes requested):
Please also delete drivers/modem/simcom-sim7080.c. This was re-added through a faulty rebase

aescolar committed:
drivers: modem: sim7080: Remove orphan file

drivers/modem/simcom-sim7080.c is not built.
And the author indicates it was added to the tree in error.
Let's remove it.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
